Sources/Misc/Signing+ResponseValidation.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Signing+ResponseValidation.swift
+//
+//  Created by Nacho Soto on 2/8/23.
+
+import Foundation
+
+extension HTTPResponse where Body == Data {
+
+    static func createAndValidate(with response: HTTPURLResponse,
+                                  body: Data,
+                                  request: HTTPRequest,
```

Comment:
Can we extend `HTTPRequest` to have a `IntegrityVerifiedHTTPRequest` that guarantees contains a Nonce? And then make this signature expect that class in particular. Doesn't make sense to validate a `HTTPRequest` that is not sending Nonces.

Reply:
That's definitely an alternative approach, but that's not very Swift-y. We'd have to move away from it being a `struct` and lose the value semantics on this type.

I think the main issue is this is bad name. It returns `HTTPResponseValidationResult.notRequested` if the request doesn't need to be verified, so maybe instead of `createAndValidate` it should just be `HTTPResponse.create(with:...)`.

Reply:
Alternatively we'd have to make `HTTPResponse` a protocol and use a generic `<Response: HTTPResponseType>` everywhere to maintain type safety at compile time instead of doing that at runtime. But at that point it would be an overkill.

Reply:
Hummm... I see... I have no clue about swift :)

And something like createRequestAndValidator() that returns a request with the nonce, and a validator ready to use (contains the nonce)? It feels weird to have this generic create and validate that sometimes will not validate, plus there is not a lot of type safety, there is no way to enforce validation for some kind of request.

That way the request remains generic, you will just build a validator when requesting a request with validation. If not requesting validation, you will just have a create that builds a request.

Again, all this not knowing a thing about swift patterns :)

Reply:
I renamed this, I think it's clearer now. I do get your point of seemingly validating requests that don't require it.
`HTTPResponse` always has an `HTTPResponseValidationResult` (even if it's `.notRequested`).
So `HTTPResponse.create(with:body:request:..)` is the new API. It may or may not do any validation, but it'll return an `HTTPResponse` with the appropriate information. Does that sound less weird?